src/elements/content-explorer/__tests__/RenameDialog.test.tsx

```diff
@@ -0,0 +1,85 @@
+    appElement: document.createElement('div'),
+    errorCode: '',
+    isLoading: false,
+    isOpen: true,
+    item: { name: 'test.txt', extension: 'txt' },
+    onCancel: jest.fn(),
+    onRename: jest.fn(),
+    parentElement: document.createElement('div'),
+};
+
+describe('elements/content-explorer/RenameDialog', () => {
+    const renderComponent = (props = {}) => render(<RenameDialog {...defaultProps} {...props} />);
+
+    test('renders the dialog with the correct initial state', () => {
+        renderComponent();
+        expect(screen.getByText('Please enter a new name for test:')).toBeInTheDocument();
+        expect(screen.getByRole('textbox')).toHaveValue('test');
+        expect(screen.getByDisplayValue('test')).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Rename' })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+    });
+
+    test('calls onCancel when cancel button is clicked', async () => {
+        renderComponent();
+        await userEvent.click(await screen.findByRole('button', { name: 'Cancel' }));
+        expect(defaultProps.onCancel).toHaveBeenCalled();
+    });
+
+    test('calls onRename with the correct values when rename button is clicked', async () => {
+        renderComponent();
+        const input = await screen.findByRole('textbox');
+        await userEvent.clear(input);
+        await userEvent.type(input, 'newname');
+        await userEvent.click(await screen.findByRole('button', { name: 'Rename' }));
+        expect(defaultProps.onRename).toHaveBeenCalledWith('newname', '.txt');
+    });
+
+    test('displays an error message when errorCode is ERROR_CODE_ITEM_NAME_IN_USE', () => {
+        renderComponent({ errorCode: 'something else' });
+        expect(screen.getByText('This name is invalid.')).toBeInTheDocument();
+    });
+
+    test('displays an error message when errorCode is ERROR_CODE_ITEM_NAME_IN_USE', () => {
+        renderComponent({ errorCode: ERROR_CODE_ITEM_NAME_IN_USE });
+        expect(screen.getByText('An item with the same name already exists.')).toBeInTheDocument();
+    });
+
+    test('displays an error message when errorCode is ERROR_CODE_ITEM_NAME_IN_USE', () => {
+        renderComponent({ errorCode: ERROR_CODE_ITEM_NAME_IN_USE });
+        expect(screen.getByText('An item with the same name already exists.')).toBeInTheDocument();
+    });
+
+    test('displays an error message when errorCode is ERROR_CODE_ITEM_NAME_TOO_LONG', () => {
+        renderComponent({ errorCode: ERROR_CODE_ITEM_NAME_TOO_LONG });
+        expect(screen.getByText('This name is too long.')).toBeInTheDocument();
+    });
+
+    test('does not call onRename if the name has not changed', async () => {
+        renderComponent();
+        await userEvent.click(screen.getByText('Rename'));
+        expect(defaultProps.onRename).not.toHaveBeenCalled();
+        expect(defaultProps.onCancel).toHaveBeenCalled();
```

Comment:
i dont think we should check a shared object like this. these tests are passing because they're relying on global jest setting `clearMocks: true` but i don't think we should rely on that and it could set up bad habits when developers work in other repos.

creating local variables like this makes the tests more independent and intentional:
```
test('...', async () => {
    const onCancel = jest.fn();
    const onRename = jest.fn();

    renderComponent({ onCancel, onRename });
    await userEvent.click(screen.getByText('Rename'));
    expect(onRename).not.toHaveBeenCalled();
    expect(onCancel).toHaveBeenCalled();
});
```